### DISCUSSION TITLE
New `--number-sections` behavior from 3.1.12 creates gap in numbering when `.unnumbered` class is used

### DISCUSSION THREAD

**Discussion by cderv** [jgm/pandoc#9513] [Q&A] [answered]
I opening this as a discussion as I am not sure from reading release note, and associated issue of commit of this is expected change or not. 

I understand Pandoc 3.1.12 is changing the behavior of `MakeSections` (03bb426a4a65918bfd6fbd63d5bb32a657df5568) but not sure how it is supposed to impact section with `.unnumbered`. They are correctly not numbered, but they now increment the counter. 

````powershell 
# pandoc 3.1.12
> pandoc --number-sections -t html
# One {-}

# Two

# Three {-}

# Four
^Z
<h1 class="unnumbered" id="one">One</h1>
<h1 data-number="2" id="two"><span
class="header-section-number">2</span> Two</h1>
<h1 class="unnumbered" id="three">Three</h1>
<h1 data-number="4" id="four"><span
class="header-section-number">4</span> Four</h1>
````

Previous behavior with 3.1.11.1 : Header with unnumbered class did not increment the section number and were skipped

````powershell
# pandoc 3.1.11.1
> pandoc --number-sections -t html
# One {-}

# Two

# Three {-}

# Four
^Z
<h1 class="unnumbered" id="one">One</h1>
<h1 data-number="1" id="two"><span
class="header-section-number">1</span> Two</h1>
<h1 class="unnumbered" id="three">Three</h1>
<h1 data-number="2" id="four"><span
class="header-section-number">2</span> Four</h1>
````

This is quite a breaking change, and I though that `.unnumbered` class could really be used to have some sections ignored in the numbering - we used this in some project (i.e **bookdown** in R Markdown ecosystem) to have Part section in HTML that where not meant to be incrementing the chapter numbers. 

This is still happening with nightly pandoc, and I found no report yet, so I want to confirm is this is really expected new behavior and we need to adapt, or if this could be unintended side effect of new 3.1.12 behavior. 

Thank you.

**Comment by jgm** [comment, marked as answer]
They shouldn't increment the counter, so it's a bug indeed. Please submit an issue.

**Reply by cderv** [reply]
Submitted as
* #9516
